pkgs/development/tools/godot/default.nix

```diff
@@ -65,39 +83,76 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
 
-  sconsFlags = "target=release_debug platform=x11";
-  preConfigure = ''
-    sconsFlags+=" ${
-      lib.concatStringsSep " "
-      (lib.mapAttrsToList (k: v: "${k}=${builtins.toJSON v}") options)
-    }"
-  '';
+  sconsFlags = let toSconsValue = x: if builtins.isBool x then lib.boolToString x else x; in
+    lib.pipe self.passthru.sconsFlagSet [
+      (lib.mapAttrsToList (k: v: "${k}=${toSconsValue v}"))
+      (lib.concatStringsSep " ")
+    ];
 
   outputs = [ "out" "dev" "man" ];
 
+  installSteps = [ "installBinStep" "installDevStep" "installManStep" "installShortcutStep" ];
+
   installPhase = ''
-    mkdir -p "$out/bin"
-    cp bin/godot.* $out/bin/godot
+    runHook preInstall
+
+    for curInstallStep in $installSteps; do
+      eval "''${!curInstallStep}"
+    done
+
+    runHook postInstall
+  '';
+
+  installBinStep = ''
+    runHook preInstallBin
+
+    bin="$out"/"$binInstallPath"
+    mkdir -p "$bin"
+    cp -R bin/. "$bin"
+    mv "$bin"/$builtGodotBinNamePattern "$bin/$installedGodotBinName"
+
+    runHook postInstallBin
+  '';
+
+  installDevStep = ''
+    runHook preInstallDev
 
     mkdir "$dev"
-    cp -r modules/gdnative/include $dev
+    cp -R modules/gdnative/include "$dev"
+
+    runHook postInstallDev
+  '';
+
+  installManStep = ''
+    runHook preInstallMan
+
+    godotman="$man"/share/man/man6
+    mkdir -p "$godotman"
+    cp misc/dist/linux/godot.6 "$godotman"
```

Comment:
Please use installManPage

Reply:
`installManPageStep` instead of `installManStep`?  I can do that.  It does make it a bit clearer.

Reply:
I mean this https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/install-shell-files.sh#L19-L50

Also this phase can be consolidated into the others.

Reply:
Ah interesting.  I just kept the logic for this step from what was there before, but rewriting in terms of this `installManPage` script should be an improvement if it works.  I'll look into it.  Thanks!